navigation/src/main/java/com/graphhopper/navigation/NavigateResponseConverter.java

```diff
@@ -136,48 +142,160 @@ private static void putLegInformation(ObjectNode legJson, ResponsePath path, int
 
+        // find out the interesting part of the intersectionDetails
         for (PathDetail intersectionDetail : intersectionDetails) {
-            if (intersectionDetail.getFirst() >= pointIndexTo) {
+            int first = intersectionDetail.getFirst();
+            if (first >= pointIndexTo) {
                 break;
             }
-            if (intersectionDetail.getFirst() >= pointIndexFrom) {
-                ObjectNode intersection = intersections.addObject();
-                Map<String, Object> intersectionValue = (Map<String, Object>) intersectionDetail.getValue();
-                // Location
-                ArrayNode locationArray = intersection.putArray("location");
-                locationArray.add(Helper.round6(points.getLon(intersectionDetail.getFirst())));
-                locationArray.add(Helper.round6(points.getLat(intersectionDetail.getFirst())));
-                // Entry
-                List<Boolean> entries = (List<Boolean>) intersectionValue.getOrDefault("entries", Collections.emptyList());
-                ArrayNode entryArray = intersection.putArray("entry");
-                for (Boolean entry : entries) {
-                    entryArray.add(entry);
+            if (first >= pointIndexFrom) {
+                list.add(intersectionDetail);
+            }
+        }
+        // nothing to be done
+        if (list.size() == 0) {
+            return list;
+        }
+
+        // Now look for adjacent intersections on same location
+        GHPoint3D intersectionPoint = points.get(list.get(0).getFirst());
+        List<Integer> duplicates = new ArrayList<>();
+        for (int i = 1; i < list.size(); i++) {
+            GHPoint3D currentIntersectionPoint = points.get(list.get(1).getFirst());
+            if (intersectionPoint.equals(currentIntersectionPoint)) {
+                duplicates.add(i - 1); // store the first index of the duplicate
+            }
+            intersectionPoint = currentIntersectionPoint;
+        }
+
+        // now iterate backwards over all duplicates, since we will remove entries from
+        // list
+        for (int dup = duplicates.size() - 1; dup >= 0; dup--) {
+            int i = duplicates.get(dup);
+            // member i and i+1 are on the same point
+            // ... -------> intersection[i].out --------> intersection[i+1].in ----------->
+            // ....
+            // Create a new PathDetail for merging both intersections into one
+            try {
+                Map<String, Object> intersection = (Map<String, Object>) list.get(i).getValue();
+
+                List<Integer> bearings = (List<Integer>) intersection.get("bearings");
+                Integer in = (Integer) intersection.get("in");
+                Integer out = (Integer) intersection.get("out");
+                List<Boolean> entry = (List<Boolean>) intersection.get("entries");
+
+                if (bearings.get(out) != 90) {
+                    // oops, we expected it to have a bearing of 90
```

Comment:
Why? Does it have to be always 90?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#atan2-double-double-
Second bullet point for Math.atan2( x-x, y-y) => 0
(note x-x = +0 according IEEE754) 

Call Chain calculateBearing / AngleCalc.ANGLE_CALC.calcAzimuth [PI/2 - 0] /  calcOrientation / Math.toDegrees / calcOrientation ...

Reply:
removing it since it seems to confuse